File: irrgarten/src/irrgarten/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author marco
 */
public class TestP1 {
    public static Dice dice = new Dice();
    
    public static int testWeapon(){
        System.out.println("TESTING WEAPON");
        
        // Test constructor without parameters
        Weapon voidWeapon = new Weapon();
        
        // Test constructor with parameters
        Weapon filledWeapon = new Weapon(dice.weaponPower() , dice.usesLeft());
        
        // Test to_String
        System.out.println("Weapons state after being created: ");
        System.out.println("Void weapon: " + voidWeapon.toString());
        System.out.println("Filled weapon: " + filledWeapon.toString() + "\n");
        
        // Test attack
        float attackPowerVoidWeapon = voidWeapon.attack();
        float attackPowerFilledWeapon = filledWeapon.attack();
        
        // State after attack
        System.out.println("State after attacking: ");
        System.out.println("Void weapon: " + voidWeapon.toString());
        System.out.println("Attack power of void weapon: " + attackPowerVoidWeapon);
        System.out.println("Filled weapon: " + filledWeapon.toString());
        System.out.println("Attack power of filled weapon: " + attackPowerFilledWeapon + "\n");

                
        // Discard method testing
        Weapon maxWeapon = new Weapon(2, 5);
        Weapon minWeapon = voidWeapon;
        Weapon averageWeapon = new Weapon(1, 3);
        System.out.println("Result discarting maxWeapon: " + maxWeapon.discard());
        System.out.println("Result discarting minWeapon: " + minWeapon.discard());
        System.out.println("Result discarting averageWeapon: " + averageWeapon.discard());
        System.out.println("\n\n");
        
        return 0;
    }
    
    public static int testShield(){
        System.out.println("TESTING SHIELD");
        
        // Test constructor without parameters
        Shield voidShield = new Shield();
        
        // Test constructor with parameters
        Shield filledShield = new Shield(dice.shieldPower() , dice.usesLeft());
        
        // Test to_String
        System.out.println("Shields state after being created: ");
        System.out.println("Void Shield: " + voidShield.toString());
        System.out.println("Filled Shield: " + filledShield.toString() + "\n");
        
        // Test Protect
        float ProtectPowerVoidShield = voidShield.protect();
        float ProtectPowerFilledShield = filledShield.protect();
        
        // State after Protect
        System.out.println("State after Protecting: ");
        System.out.println("Void Shield: " + voidShield.toString());
        System.out.println("Protect power of void Shield: " + ProtectPowerVoidShield);
        System.out.println("Filled Shield: " + filledShield.toString());
        System.out.println("Protect power of filled Shield: " + ProtectPowerFilledShield + "\n");
        
        // Discard method testing
        Shield maxShield = new Shield(2, 5);
        Shield minShield = voidShield;
        Shield averageShield = new Shield(1, 3);
        System.out.println("Result discarting maxShield: " + maxShield.discard());
        System.out.println("Result discarting minShield: " + minShield.discard());
        System.out.println("Result discarting averageShield: " + averageShield.discard());
        System.out.println("\n\n");
        
        return 0;
    }
    
    
    public static int testGameState(){
        System.out.println("TESTING GAMESTATE");
        
        GameState gs = new GameState("Labyrinth state", "Players",
                                    "Monsters", 0, true,
                                    "Log");
        System.out.println("Laberynth state: " + gs.getLabyrinthv());
        System.out.println("Players: " + gs.getPlayers());
        System.out.println("Monsters: " + gs.getMonsters());
        System.out.println("Current Player: " + gs.getCurrentPlayer());
        System.out.println("Winner: " + gs.isWinner());
        System.out.println("Log: " + gs.getLog());
        System.out.println("\n\n");
        
        return 0;
    }
    
    public static int testEnums(){
        System.out.println("TESTING ENUMS");
        
        GameCharacter gc = GameCharacter.PLAYER;
        Directions dir = Directions.UP;
        Orientation ori = Orientation.HORIZONTAL;
        
        System.out.println("Game Character: " + gc);
        System.out.println("Directions: " + dir);
        System.out.println("Orientation: " + ori);
        System.out.println("\n\n");
        
        return 0;
    }
    
    public static int testDice(){
        System.out.println("TESTING DICE");

        float averageProbUses = 0;
        float averageProbIntelligence = 0;
        float averageProbStrength = 0;
        float averageProbResurrect = 0;
        float averageProbWeaponsReward = 0;
        float averageProbShieldsReward = 0;
        float averageProbHealthReward = 0;
        float averageProbAttack = 0;
        float averageProbProtect = 0;
        float averageProbIntensity = 0;
    
        int n = 10000;
        
        for(int i=0; i<n; i++){
            averageProbUses += dice.usesLeft();
            averageProbIntelligence += dice.randomIntelligence();
            averageProbStrength += dice.randomStrength();
            averageProbResurrect += (dice.resurrectPlayer() == true) ? 1 : 0;
            averageProbWeaponsReward += dice.weaponsReward();
            averageProbShieldsReward += dice.shieldsReward();
            averageProbHealthReward += dice.healthReward();
            averageProbAttack += dice.weaponPower();
            averageProbProtect += dice.shieldPower();
            averageProbIntensity += dice.intensity(10);
        }
        
        averageProbUses /= n*5;
        averageProbIntelligence /= n*10.0f;
        averageProbStrength /= n*10.0f;
        averageProbResurrect /= n*1;
        averageProbWeaponsReward /= n*2;
        averageProbShieldsReward /= n*3;
        averageProbHealthReward /= n*5;
        averageProbAttack /= n*3;
        averageProbProtect /= n*2;
        averageProbIntensity /= n*10;
        
        System.out.println( "Average Prob Uses :" + averageProbUses*100 + "%");
        System.out.println( "Average Prob Intelligence :" + averageProbIntelligence*100 + "%");
        System.out.println( "Average Prob Strength :" + averageProbStrength*100 + "%");
        System.out.println( "Average Prob Resurrect :" + averageProbResurrect*100 + "%");
        System.out.println( "Average Prob Weapons Reward :" + averageProbWeaponsReward*100 + "%");
        System.out.println( "Average Prob Shields Reward :" + averageProbShieldsReward*100 + "%");
        System.out.println( "Average Prob HealthReward :" + averageProbHealthReward*100 + "%");
        System.out.println( "Average Prob Attack :" + averageProbAttack*100 + "%");
        System.out.println( "Average Prob Protect :" + averageProbProtect*100 + "%");
        System.out.println( "Average Prob Intensity :" + averageProbIntensity*100 + "%");
        System.out.println("\n\n");
        
        return 0;
    }
    
    public static void main(String[] args){
        
        int status = testWeapon() + testShield() + testGameState() + 
                    testEnums() + testDice();
        
        System.exit(status);
    }
}

File: PDFs/test_p1.rb
# frozen_string_literal: true
require_relative '../irrgarten_ruby/dice'
require_relative 'weapon'
require_relative 'shield'
require_relative 'game_character'
require_relative 'orientation'
require_relative 'directions'
require_relative 'game_state'

class TestP1
  
  def self.test_weapon
    puts "TESTING WEAPON"

    void_weapon = Weapon.new(0,0)
    filled_weapon = Weapon.new(Dice.weapon_power, Dice.uses_left)

    puts "Weapons state after being created: "
    puts "Void weapon: #{void_weapon.to_s}"
    puts "Filled weapon: #{filled_weapon.to_s}"

    attack_power_void_weapon = void_weapon.attack
    attack_power_filled_weapon = filled_weapon.attack

    puts "State after attacking: "
    puts "Void weapon: #{void_weapon.to_s}"
    puts "Attack power of void weapon: #{attack_power_void_weapon}"
    puts "Filled weapon: #{filled_weapon.to_s}"
    puts "Attack power of filled weapon: #{attack_power_filled_weapon.to_s}"

    max_weapon = Weapon.new(2,5)
    min_weapon = void_weapon
    average_weapon = Weapon.new(1,3)

    puts "Result discarding max_weapon: #{max_weapon.discard}"
    puts "Result discarding min_weapon: #{min_weapon.discard}"
    puts "Result discarding average_weapon: #{average_weapon.discard}"
    puts "\n\n"

    0
  end

  def self.test_shield
    puts "TESTING SHIELD"

    void_shield = Shield.new(0,0)
    filled_shield = Shield.new(Dice.shield_power, Dice.uses_left)

    puts "Shields state after being created: "
    puts "Void shield: #{void_shield.to_s}"
    puts "Filled shield: #{filled_shield.to_s}"

    protect_power_void_shield = void_shield.protect
    protect_power_filled_shield = filled_shield.protect

    puts "State after protecting: "
    puts "Void shield: #{void_shield.to_s}"
    puts "Attack power of void shield: #{protect_power_void_shield}"
    puts "Filled shield: #{filled_shield.to_s}"
    puts "Attack power of filled shield: #{protect_power_filled_shield.to_s}"

    max_shield = Shield.new(2,5)
    min_shield = void_shield
    average_shield = Shield.new(1,3)

    puts "Result discarding max_shield: #{max_shield.discard}"
    puts "Result discarding min_shield: #{min_shield.discard}"
    puts "Result discarding average_shield: #{average_shield.discard}"
    puts "\n\n"

    0
  end

  def self.test_game_state
    puts "TESTING GAME STATE"

    gs = GameState.new("Okay", "Antonio, Pepe", "Godzilla", 3, true, "Log")
    puts "Labyrinth state: #{gs.labyrinthv}"
    puts "Players: #{gs.players}"
    puts "Monsters: #{gs.monsters}"
    puts "Current Players: #{gs.current_player}"
    puts "Winner: #{gs.winner}"
    puts "Log: #{gs.log}"
    puts "\n\n"

    0
  end

  def self.test_enums
    puts "TESTING ENUMS"

    gc = GameCharacter::PLAYER
    dir = Directions::UP
    ori = Orientation::HORIZONTAL

    puts "Game Character: #{gc}"
    puts "Directions: #{dir}"
    puts "Orientation: #{ori}"
    puts "\n\n"

    0
  end

  def self.test_dice
    puts "TESTING Dice"

    average_prob_uses = 0
    average_prob_intelligence = 0
    average_prob_strength = 0
    average_prob_resurrect = 0
    average_prob_weapons_reward = 0
    average_prob_shields_reward = 0
    average_prob_health_reward = 0
    average_prob_attack = 0
    average_prob_protect = 0
    average_prob_intensity = 0

    n = 100000

    n.times do
      average_prob_uses += Dice.uses_left
      average_prob_intelligence += Dice.random_intelligence
      average_prob_strength += Dice.random_strength
      average_prob_resurrect += (Dice.resurrect_player) ? 1 : 0
      average_prob_weapons_reward += Dice.weapons_reward
      average_prob_shields_reward += Dice.shields_reward
      average_prob_health_reward += Dice.health_reward
      average_prob_attack += Dice.weapon_power
      average_prob_protect += Dice.shield_power
      average_prob_intensity += Dice.intensity(10)
    end

    average_prob_uses /= n*5.0
    average_prob_intelligence /= n*10.0
    average_prob_strength /= n*10.0
    average_prob_resurrect /= n*1.0
    average_prob_weapons_reward /= n*2.0
    average_prob_shields_reward /= n*3.0
    average_prob_health_reward /= n*5.0
    average_prob_attack /= n*3.0
    average_prob_protect /= n*2.0
    average_prob_intensity /= n*10.0

    puts "Average Prob Uses :  #{average_prob_uses*100}%"
    puts "Average Prob Intelligence :  #{average_prob_intelligence*100}%"
    puts "Average Prob Strength :  #{average_prob_strength*100}%"
    puts "Average Prob Resurrect :  #{average_prob_resurrect*100}%"
    puts "Average Prob Weapons Reward :  #{average_prob_weapons_reward*100}%"
    puts "Average Prob Shields Reward :  #{average_prob_shields_reward*100}%"
    puts "Average Prob HealthReward :  #{average_prob_health_reward*100}%"
    puts "Average Prob Attack :  #{average_prob_attack*100}%"
    puts "Average Prob Protect :  #{average_prob_protect*100}%"
    puts "Average Prob Intensity :  #{average_prob_intensity*100}%"
    puts "\n\n"

    0
  end

  def self.main
    test_weapon + test_shield + test_game_state + test_enums + test_dice
  end
end

TestP1.main
